docs/how-clerk-works/overview.mdx

```diff
@@ -0,0 +1,256 @@
+
+## Stateful authentication
+
+To understand how authentication works in Clerk, it's important to first understand how the most common implementation of authentication logic works: the traditional "stateful authentication" model, also known as "session token authentication".
+
+A user's process of signing in would work as follows:
+
+1. The user initiates authentication by navigating to `example.com/sign-in`, entering their credentials (e.g. username/password), and submitting the form, usually by clicking a "submit" button. This makes a request to the server with the credentials.
+1. The server validates the credentials against a database. This is normally done by [hashing](TODO) the password and comparing it with a stored password hash. Upon successful validation, it creates a new [session](TODO) in the database associated with the user. The session represents an instance in which a given user has signed in, and typically contains information such as a session ID, the user's ID, an expiration time, and session status (active, revoked, expired, etc.).
+1. The server responds to the browser's request by setting the session ID in a [`Set-Cookie`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie) header in the response. This cookie will be automatically included in future requests from the browser in order to authenticate the user.
+   <Video
+     src="/docs/images/how-clerk-works/stateful-auth.mp4"
+     width="1600"
+     height="900"
+     autoPlay
+     muted
+     loop
+     playsInline
+   />
+1. The next time the browser sends a request to the server, it automatically includes the session cookie. The server checks the database for the session ID and retrieves the associated user ID and session metadata. If the session is valid and active, the server has verified that the user has authenticated, andcan then use the user ID to fetch any required user data and process the request.
+   <Video
+     src="/docs/images/how-clerk-works/stateful-auth-2.mp4"
+     width="1600"
+     height="900"
+     autoPlay
+     muted
+     loop
+     playsInline
+   />
+
+> [!NOTE]
+> If an attacker obtains your session ID, they can fully you as a user and take over your account. To mitigate this risk, Clerk implements the following security measures: 1. [HTTPS](https://developer.mozilla.org/en-US/docs/Glossary/HTTPS) is enforced to prevent [man-in-the-middle attacks](https://en.wikipedia.org/wiki/Man-in-the-middle_attack) that could intercept session IDs in transit, and 2. the [`HttpOnly` flag](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#httponly) is set on session cookies to prevent malicious JavaScript from accessing them through cross-site scripting (XSS) attacks. [Read more about Clerk's security measures](/docs/security/overview).
```

Comment:
This section needs to be heavily modified - we're not talking about Clerk's auth scheme here yet so an injection about how Clerk does auth feels out of place. Also, we don't actually use httponly for session cookies, so this isn't accurate. Will take another pass on this later but wanted to drop a note so i dont forget at least

Reply:
oh, i wrote that incorrectly about the `httponly` flag :(
and i see, good point about injecting how clerk handles it! i thought we were mentioning things about other flows (in this case, stateful) in order to compare to clerk's flows. 
i would ask then, what is the purpose of providing this information? why provide this callout for users?